### DISCUSSION TITLE
Installing cartopy in the Jupyter notebook

### DISCUSSION THREAD

**Discussion by erkoj** [pydata/xarray#5069] [Q&A] [answered]
Hello,
I have tried to install cartopy in my Win10 Jupyter notebook using:  !{sys.executable} -m pip install cartopy

Unfortunately, it gives error. I understand that there is problem with GEOS, but how to solve it?

The error code is very long, I copy here only the beginning and last part:

ollecting cartopy
  Using cached https://files.pythonhosted.org/packages/46/c1/04e50c9986842f00f7db0e7a65caa896840050d7328f74e5b7437aa01179/Cartopy-0.18.0.tar.gz
Requirement already satisfied: numpy>=1.10 in c:\users\erkok\appdata\local\continuum\anaconda3\lib\site-packages (from cartopy) (1.20.1)
Requirement already satisfied: shapely>=1.5.6 in c:\users\erkok\appdata\local\continuum\anaconda3\lib\site-packages (from cartopy) (1.7.1)
Requirement already satisfied: pyshp>=1.1.4 in c:\users\erkok\appdata\local\continuum\anaconda3\lib\site-packages (from cartopy) (2.0.1)
Requirement already satisfied: six>=1.3.0 in c:\users\erkok\appdata\local\continuum\anaconda3\lib\site-packages (from cartopy) (1.12.0)
Requirement already satisfied: setuptools>=0.7.2 in c:\users\erkok\appdata\local\continuum\anaconda3\lib\site-packages (from cartopy) (54.2.0)
Building wheels for collected packages: cartopy
  Running setup.py bdist_wheel for cartopy: started
  Running setup.py bdist_wheel for cartopy: finished with status 'error'
  Complete output from command C:\Users\erkok\AppData\Local\Continuum\anaconda3\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\erkok\\AppData\\Local\\Temp\\pip-install-c35y7rgn\\cartopy\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d C:\Users\erkok\AppData\Local\Temp\pip-wheel-jyushzfq --python-tag cp37:
  C:\Users\erkok\AppData\Local\Temp\pip-install-c35y7rgn\cartopy\setup.py:107: UserWarning: Unable to determine GEOS version. Ensure you have 3.3.3 or later installed, or installation may fail.
    '.'.join(str(v) for v in GEOS_MIN_VERSION), ))

---------------------

----------------------
building 'cartopy.trace' extension
    creating build\temp.win-amd64-3.7
    creating build\temp.win-amd64-3.7\Release
    creating build\temp.win-amd64-3.7\Release\lib
    creating build\temp.win-amd64-3.7\Release\lib\cartopy
    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -I. -I./lib/cartopy -IC:\Users\erkok\AppData\Local\Continuum\anaconda3\lib\site-packages\numpy\core\include -IC:\Users\erkok\AppData\Local\Continuum\anaconda3\include -IC:\Users\erkok\AppData\Local\Continuum\anaconda3\include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\8.1\include\shared" "-IC:\Program Files (x86)\Windows Kits\8.1\include\um" "-IC:\Program Files (x86)\Windows Kits\8.1\include\winrt" /EHsc /Tplib/cartopy/trace.cpp /Fobuild\temp.win-amd64-3.7\Release\lib/cartopy/trace.obj
    trace.cpp
    lib/cartopy/trace.cpp(665): fatal error C1083: Cannot open include file: 'geos_c.h': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\cl.exe' failed with exit status 2
    
    ----------------------------------------

  Failed building wheel for cartopy
Command "C:\Users\erkok\AppData\Local\Continuum\anaconda3\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\erkok\\AppData\\Local\\Temp\\pip-install-c35y7rgn\\cartopy\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\erkok\AppData\Local\Temp\pip-record-va2i27q2\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\erkok\AppData\Local\Temp\pip-install-c35y7rgn\cartopy\

**Comment by mathause** [comment, marked as answer]
cartopy has a dependecy that cannot be installed with pip. So the recommendation is to use conda.

If you need help installing cartopy with pip please check the cartopy ressources.

**Comment by erkoj** [comment]
Great - I wasted many hours trying, reading etc and now it just works:)
